src/highlight.js

```diff
@@ -513,16 +513,24 @@ https://highlightjs.org/
     compileMode(language);
   }
 
-  /*
-  Core highlighting function. Accepts a language name, or an alias, and a
-  string with the code to highlight. Returns an object with the following
-  properties:
-
-  - relevance (int)
-  - value (an HTML string with highlighting markup)
 
+  /**
+   * Core highlighting function.
+   *
+   * @param {string} languageName - the language to use for highlighting
+   * @param {string} code - the code to highlight
+   * @param {boolean} ignore_illegals - whether to ignore illegal matches, default is to bail
+   * @param {array<mode>} continuation - array of continuation modes
+   *
+   * @returns an object that represents the result
+   * @property {string} language - the language name
+   * @property {number} relevance - the relevance score
+   * @property {string} value - the highlighted HTML code
+   * @property {mode} top - top of the current mode stack
+   * @property {boolean} illegal - indicates whether any illegal matches were found
   */
-  function highlight(name, value, ignore_illegals, continuation) {
+  function highlight(languageName, code, ignore_illegals, continuation) {
+    var codeToHighlight = code; // for search and replace
```

Comment:
`codeToHighlight` is fine, but again, why not simply
```js
function highlight(languageName, codeToHighlight, ignore_illegals, continuation) {
```
?
Doesn't renaming by assigning the value to another variable look strange?

Reply:
I see what you're getting at, but doesn't bother me. In the method definition all that wording is extraneous... highlight(codeToHighlight) is silly and repetitive.  But in the huge scope this variable is used in the longer naming makes a lot more sense.

So this is the best of both worlds IMHO.

If you insist I'll change it, but I'd rather not. :-)

Reply:
Okay, agree. But let's remove `// for search and replace`?

Reply:
LOL, I thought that would appease you... I'll remove.

Reply:
I think it's rather more confusing (: